src/service-implementations/lf-repo-tree-node-service.ts

```diff
@@ -409,34 +419,15 @@ export class LfRepoTreeNodeService implements LfTreeNodeService {
     }
   }
 
-  private createFolderNode(entry: Entry, repoName: string, parent?: LfRepoTreeNode): LfRepoTreeNode {
+  private createFolderNodeWithParent(entry: Entry, parent: LfRepoTreeNode): LfRepoTreeNode {
     if (!entry.id) {
       throw new Error('entryId is undefined');
     }
-    if (parent && !parent.path) {
-      throw new Error('path is undefined on parent entry');
-    }
     let entryName: string | undefined = entry.name;
-    let path: string;
-    if (entry.id === 1) {
-      // rootNode does not have a name, default into repoName
-      path = '\\';
-      if (!entryName || entryName.length === 0) {
-        entryName = repoName;
-      }
-    }
-    else {
-      if (!entryName || entryName.length === 0) {
-        entryName = entry.id.toString();
-      }
-      path = parent ? PathUtils.combinePaths(parent.path, entryName) : entry.fullPath;
-      if (!path) {
-        throw new Error('fullPath is undefined');
-      }
-      else if (path.endsWith('\\') && path.length > 1) {
-        path = path.slice(0, -1);
-      }
+    if (!entryName || entryName.length === 0) {
+      entryName = entry.id.toString();
     }
+    const path = this.getFullPath(parent, entryName);
 
```

Comment:
can actually call getFolderTreeNodeWithPath at this point so we don't have to assign each of the properties again?